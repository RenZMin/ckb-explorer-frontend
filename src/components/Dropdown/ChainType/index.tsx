import i18n from '../../../utils/i18n'
import CONFIG from '../../../config'
import { isMainnet } from '../../../utils/chain'
import { ChainTypePanel } from './styled'
import SimpleButton from '../../SimpleButton'

const testnetUrl = `${CONFIG.MAINNET_URL}/${CONFIG.TESTNET_NAME}`

export default ({ setShow, left, top }: { setShow: Function; left: number; top: number }) => {
  const hideDropdown = () => {
    setShow(false)
  }

  return (
    <ChainTypePanel left={left} top={top} onMouseLeave={hideDropdown}>
      <SimpleButton className={`chain__type__${isMainnet() ? 'selected' : 'normal'}`} onClick={hideDropdown}>
        <a href={CONFIG.MAINNET_URL}>{i18n.t('blockchain.mainnet')}</a>
      </SimpleButton>
      <div className="chain__type__separate" />
      <SimpleButton className={`chain__type__${!isMainnet() ? 'selected' : 'normal'}`} onClick={hideDropdown}>
        <a href={testnetUrl}>{`${CONFIG.TESTNET_NAME} ${i18n.t('blockchain.testnet')}`}</a>
      </SimpleButton>
    </ChainTypePanel>
  )
}
