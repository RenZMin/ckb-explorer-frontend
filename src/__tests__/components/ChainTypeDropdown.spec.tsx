import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ChainType from '../../components/Dropdown/ChainType'

describe('ChainType Dropdown Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <ChainType setShow={() => {}} top={0} left={0} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.chain__type__normal')).toHaveLength(1)
    expect(wrapper.find('.chain__type__selected')).toHaveLength(1)
  })
})
