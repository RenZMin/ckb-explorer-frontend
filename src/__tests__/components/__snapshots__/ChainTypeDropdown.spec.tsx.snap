// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainType Dropdown Component shallow renders 1`] = `
<div
  className="sc-bdnxRM ruRUW"
  onMouseLeave={[Function]}
>
  <div
    className="sc-gtsrHT kcdbmG chain__type__selected"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="button"
    tabIndex={-1}
  >
    <a
      href="http://localhost:3000"
    >
      Lina Mainnet
    </a>
  </div>
  <div
    className="chain__type__separate"
  />
  <div
    className="sc-gtsrHT kcdbmG chain__type__normal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseOver={[Function]}
    role="button"
    tabIndex={-1}
  >
    <a
      href="http://localhost:3000/testnet"
    >
      testnet Testnet
    </a>
  </div>
</div>
`;
